src/Core/OrganizationFeatures/OrganizationSubscriptions/UpgradeOrganizationPlanCommand.cs

```diff
@@ -216,9 +216,19 @@ await _organizationUserRepository.GetManyByOrganizationAsync(organization.Id,
 
         if (string.IsNullOrWhiteSpace(organization.GatewaySubscriptionId))
         {
-            paymentIntentClientSecret = await _paymentService.UpgradeFreeOrganizationAsync(organization,
-                newPlan, upgrade);
-            success = string.IsNullOrWhiteSpace(paymentIntentClientSecret);
+            try
+            {
+                paymentIntentClientSecret = await _paymentService.UpgradeFreeOrganizationAsync(organization,
+                    newPlan, upgrade);
+                success = string.IsNullOrWhiteSpace(paymentIntentClientSecret);
+            }
+            catch
+            {
+                await _paymentService.CancelAndRecoverChargesAsync(organization);
+                organization.GatewayCustomerId = null;
+                await _organizationService.ReplaceAndUpdateCacheAsync(organization);
+                throw;
```

Comment:
❓ Is there a reason we're re-throwing here rather than setting `success` to `false`?

Reply:
If you setting success as  false, it will upgrade the organization to a new plan but while the plan upgrade in stripe failed. And User will not get to see the reason for failure.